File: gameSrc/classes/uiCustomField.js
/**
 *
 * @class
 */
cwt.UICustomField = my.Class( null, cwt.UIField, /** @lends cwt.UIField.prototype */ {

  constructor: function (x, y, w, h, key, drawFn) {
    cwt.UIField.call(this,x, y, w, h, key, 0, cwt.UIField.STYLE_NORMAL);
    this.text = "";
    this.draw = drawFn;
  }
});

File: gameSrc/net/wolfTec/wtEngine/uiWidgets/UiCustomField.java
package net.wolfTec.wtEngine.uiWidgets;

import org.stjs.javascript.dom.canvas.CanvasRenderingContext2D;
import org.stjs.javascript.functions.Callback1;

public class UiCustomField extends UiField {

    private final Callback1<CanvasRenderingContext2D> drawFn;

    public UiCustomField(int x, int y, int w, int h, String key, Callback1<CanvasRenderingContext2D> drawFn) {
        super(x, y, w, h, key, 0, UiField.STYLE_NORMAL, null);
        this.text = "";
        this.drawFn = drawFn;
    }

    @Override
    public void draw(CanvasRenderingContext2D ctx) {
        drawFn.$invoke(ctx);
    }
}
